rskj-core/src/test/java/co/rsk/validators/BlockTxsValidationRuleTest.java

```diff
@@ -0,0 +1,124 @@
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+package co.rsk.validators;
+
+import co.rsk.core.RskAddress;
+import co.rsk.db.RepositoryLocator;
+import co.rsk.db.RepositorySnapshot;
+import org.bouncycastle.util.BigIntegers;
+import org.ethereum.TestUtils;
+import org.ethereum.core.Block;
+import org.ethereum.core.BlockHeader;
+import org.ethereum.core.Transaction;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.math.BigInteger;
+import java.util.Arrays;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class BlockTxsValidationRuleTest {
+    private BlockTxsValidationRule rule;
+    private RepositorySnapshot repositorySnapshot;
+    private Block parent;
+
+    @Before
+    public void setup() {
+        parent = mock(Block.class);
+        BlockHeader parentHeader = mock(BlockHeader.class);
+        when(parent.getHeader()).thenReturn(parentHeader);
+
+        RepositoryLocator repositoryLocator = mock(RepositoryLocator.class);
+        repositorySnapshot = mock(RepositorySnapshot.class);
+        when(repositoryLocator.snapshotAt(parentHeader)).thenReturn(repositorySnapshot);
+
+        rule = new BlockTxsValidationRule(repositoryLocator);
+    }
+
+    @Test
+    public void validNonce() {
+        RskAddress sender = TestUtils.randomAddress();
+        initialNonce(sender, 42);
+
+        Block block = block(transaction(sender, 42));
+
+        Assert.assertTrue(rule.isValid(block, parent));
+    }
+
+    @Test
+    public void validTransactionsWithSameNonceAndDifferentSenders() {
+        RskAddress sender1 = TestUtils.randomAddress();
+        initialNonce(sender1, 64);
+        RskAddress sender2 = TestUtils.randomAddress();
+        initialNonce(sender2, 64);
+
+        Block block = block(transaction(sender1, 64), transaction(sender2, 64));
+
+        Assert.assertTrue(rule.isValid(block, parent));
+    }
+
+    @Test
+    public void invalidNonceLower() {
+        RskAddress sender = TestUtils.randomAddress();
+        initialNonce(sender, 42);
+
+        Block block = block(transaction(sender, 41));
+
+        Assert.assertFalse(rule.isValid(block, parent));
+    }
+
+    @Test
+    public void invalidNonceHigher() {
+        RskAddress sender = TestUtils.randomAddress();
+        initialNonce(sender, 42);
+
+        Block block = block(transaction(sender, 43));
+
+        Assert.assertFalse(rule.isValid(block, parent));
+    }
+
+    @Test
+    public void invalidTransactionsWithSameNonceAndSameSender() {
+        RskAddress sender = TestUtils.randomAddress();
+        initialNonce(sender, 42);
+
+        Block block = block(transaction(sender, 42), transaction(sender, 42));
+
+        Assert.assertFalse(rule.isValid(block, parent));
+    }
+
```

Comment:
I would add a test where a list of txs are checked.
- (42, 43) should be valid
- (42, 44) should be invalid

Reply:
Both added and passing :)